### ISSUE TITLE
[3.12.1] Error in database.log

### ISSUE THREAD

**Issue by skodkim** [Thalassicus/cep-bnw#110]
3.12.1 introduced a new error in database.log, which appears twice: "no such column: Cost"

Don't know how important this is since I haven't played this version yet.

Log attached

\Skodkim
![database log](https://f.cloud.github.com/assets/6018459/1833068/ca2b9cac-73c2-11e3-99b7-922165b152d1.jpg)

**Comment by stackpoint** [comment]
No such column sounds like a table creation issue but I couldn't find anything in there to indicate a problem. I'll have to check if I can duplicate this error.

**Comment by GrantSP** [comment]
The log still has the "Cost" listing but since I can't see any in game errors yet perhaps we could leave this active to see if we can connect it with something later.

**Comment by stackpoint** [comment]
I believe the error is with the commit here: https://github.com/Thalassicus/cep-bnw/commit/18ee2100bae11b4110de292bed2768cba67aaeca#diff-8d42450a3eff6b4f1e36d39f977d12d8

The table Building_Flavors does not have a Cost column. I'm not familiar enough with SQL to fix this. You need to call the Cost column from the Buildings table with the correct row.

**Comment by GrantSP** [comment]
Not sure what the intent of that sql command is.

There is no need to DELETE anything from _Building_Flavors_.
If on the other hand the dummy buildings are being deleted based on the _Buildings_ table I can't see the logic there also. ;(

Thal will need to look at this and decide.

**Comment by stackpoint** [comment]
The idea behind the code is to delete any flavors from buildings that have a production cost of 0 or -1. This should mostly affect dummy buildings. However, this would(?) affect religion only buildings too and I'm not sure if they require flavors.

**Comment by GrantSP** [comment]
Yeah but the point I am making is there are none of the targeted buildings in the _Building_Flavors_ table.
Deleting from the _Buildings_ table would be disastrous, so that can't be the area where the deletions are to take place.

**Comment by stackpoint** [comment]
I'm not 100% sure about my sql but I believe this is what the fixed code should look like:

``` sql
DELETE FROM Building_Flavors WHERE BuildingType IN (
    SELECT Type FROM Buildings
    WHERE NOT IsVisible OR NOT (Cost > 0 OR FaithCost > 0)
);
```

I've added the condition that it must not have a build cost and a faith cost or be invisible to be deleted from the building_flavors table. If this checks out I'll replace the relevant code.

**Comment by stackpoint** [comment]
Nevermind. It looks like Thal fixed this in the 3.13 release but had not committed it to GitHub.
